core/src/storage2/collections/binary_heap/tests.rs

```diff
@@ -0,0 +1,378 @@
+    assert_eq!(heap.peek(), Some(&10));
+    {
+        let mut top = heap.peek_mut().unwrap();
+        *top -= 2;
+    }
+    assert_eq!(heap.peek(), Some(&9));
+}
+
+#[test]
+fn peek_mut_pop_works() {
+    let data = vec![2, 4, 6, 2, 1, 8, 10, 3, 5, 7, 0, 9, 1];
+    let mut heap = heap_from_slice(&data);
+    assert_eq!(heap.peek(), Some(&10));
+    {
+        let mut top = heap.peek_mut().unwrap();
+        *top -= 2;
+        assert_eq!(PeekMut::pop(top), 8);
+    }
+    assert_eq!(heap.peek(), Some(&9));
+}
+
+#[test]
+fn min_heap_works() {
+    let data = vec![2, 4, 6, 2, 1, 8, 10, 3, 5, 7, 0, 9, 1]
+        .iter()
+        .map(|x| Reverse::new(*x))
+        .collect::<Vec<_>>();
+    let mut sorted = data.clone();
+    sorted.sort();
+    let mut heap = heap_from_slice(&data);
+    while !heap.is_empty() {
+        assert_eq!(heap.peek().unwrap(), sorted.last().unwrap());
+        assert_eq!(heap.pop().unwrap(), sorted.pop().unwrap());
+    }
+}
+
+#[test]
+fn spread_layout_push_pull_works() -> env::Result<()> {
+    env::test::run_test::<env::DefaultEnvTypes, _>(|_| {
+        let heap1 = heap_from_slice(&[b'a', b'b', b'c', b'd']);
+        let root_key = Key::from([0x42; 32]);
+        SpreadLayout::push_spread(&heap1, &mut KeyPtr::from(root_key));
+        // Load the pushed binary heap into another instance and check that
+        // both instances are equal:
+        let heap2 =
+            <BinaryHeap<u8> as SpreadLayout>::pull_spread(&mut KeyPtr::from(root_key));
+        assert_eq!(heap1, heap2);
+        Ok(())
+    })
+}
+
+#[test]
+#[should_panic(expected = "encountered empty storage cell")]
+fn spread_layout_clear_works() {
+    env::test::run_test::<env::DefaultEnvTypes, _>(|_| {
+        let heap1 = heap_from_slice(&[b'a', b'b', b'c', b'd']);
+        let root_key = Key::from([0x42; 32]);
+        SpreadLayout::push_spread(&heap1, &mut KeyPtr::from(root_key));
+        // It has already been asserted that a valid instance can be pulled
+        // from contract storage after a push to the same storage region.
+        //
+        // Now clear the associated storage from `heap1` and check whether
+        // loading another instance from this storage will panic since the
+        // heap's length property cannot read a value:
+        SpreadLayout::clear_spread(&heap1, &mut KeyPtr::from(root_key));
+        let _ =
+            <BinaryHeap<u8> as SpreadLayout>::pull_spread(&mut KeyPtr::from(root_key));
+        Ok(())
+    })
+    .unwrap()
+}
+
+#[test]
+#[should_panic(expected = "encountered empty storage cell")]
+fn drop_works() {
+    env::test::run_test::<env::DefaultEnvTypes, _>(|_| {
+        let root_key = Key::from([0x42; 32]);
+        {
+            let heap = heap_from_slice(&[23, 25, 65]);
+            SpreadLayout::push_spread(&heap, &mut KeyPtr::from(root_key));
+
+            let _ = <BinaryHeap<u8> as SpreadLayout>::pull_spread(&mut KeyPtr::from(
+                root_key,
+            ));
```

Comment:
This test could also panic here, resulting in a false positive. I would add an assertion after it to prevent this from happening.